deepcell_toolbox/deep_watershed.py

```diff
@@ -161,5 +187,88 @@ def deep_watershed_mibi(outputs,
+            - fgbg-fg: prediction for the foreground/background transform
+            - pixelwise-interior: Prediction for the interior/border/background transform.
+        compartment: which cellular compartments to generate predictions for.
+            must be one of 'whole_cell', 'nuclear', 'both'
+        whole_cell_kwargs (dict): Optional list of post-processing kwargs for whole-cell prediction
+        nuclear_kwargs (dict): Optional list of post-processing kwargs for nuclear prediction
+
+    Returns:
+        numpy.array: Uniquely labeled mask for each compartment
+
+    Raises:
+        ValueError: for invalid compartment flag
+    """
+
+    valid_compartments = ['whole-cell', 'nuclear', 'both']
+
+    if whole_cell_kwargs is None:
+        whole_cell_kwargs = {}
+
+    if nuclear_kwargs is None:
+        nuclear_kwargs = {}
+
+    if compartment not in valid_compartments:
+        raise ValueError('Invalid compartment supplied: {}. '
+                         'Must be one of {}'.format(compartment, valid_compartments))
+
+    if compartment == 'whole-cell':
+        label_images = deep_watershed_mibi(model_output=model_output['whole-cell'],
+                                           **whole_cell_kwargs)
+    elif compartment == 'nuclear':
+        label_images = deep_watershed_mibi(model_output=model_output['nuclear'],
+                                           **nuclear_kwargs)
+    else:
+        label_images_cell = deep_watershed_mibi(model_output=model_output['whole-cell'],
+                                                **whole_cell_kwargs)
+
+        label_images_nucleus = deep_watershed_mibi(model_output=model_output['nuclear'],
+                                                   **nuclear_kwargs)
+
+        label_images = np.concatenate((label_images_cell, label_images_nucleus), axis=-1)
 
     return label_images
+
+
+def format_output_multiplex(output_list):
+    """Takes list of model outputs and formats into a dictionary for better readability
+
+    Args:
+        output_list: list of predictions from semantic heads
+
+    Returns:
+        formatted_dict: dictionary with predictions
+
+    Raises: ValueError if model output list is not len(8)
+    """
+
+    if len(output_list) != 8:
+        raise ValueError('output_list was length {}, expecting length 8'.format(len(output_list)))
+
+    formatted_dict = {'whole-cell': {'inner-distance': output_list[0],
+                                     'outer-distance': output_list[1],
+                                     'fgbg-fg': output_list[2][..., :1],
+                                     'pixelwise-interior': output_list[3][..., 1:2]},
+                      'nuclear': {'inner-distance': output_list[4],
+                                  'outer-distance': output_list[5],
+                                  'fgbg-fg': output_list[6][..., :1],
+                                  'pixelwise-interior': output_list[7][..., 1:2]}
+                      }
```

Comment:
Also, just curious, what's up with the slicing in fgbg and pixelwise?

Reply:
I did that so that each semantic head produces a single [batch x rows x cols x 1] array that can be used for watershed. We always use the foreground prediction from fgbg, and the interior prediction from pixelwise, for the postprocessing. That way they can be swapped in by specifying a different maxima_model name without any code changes